common/ctree.h

```diff
@@ -336,4 +398,13 @@ class ctree
 	node_base_t* root_node;
 };
 
+inline typename std::array<uint64_t, 4>& operator+=(std::array<uint64_t, 4>& a, const std::array<uint64_t, 4>& b)
+{
+	for (std::size_t i = 0, e = a.size(); i < e; ++i)
+	{
+		a[i] += b[i];
+	}
+	return a;
+}
```

Comment:
Why four specifically?
Can't we just do smth like
```cpp
template <std::size_t N>
inline std::array<uint64_t, N>& operator+=(std::array<uint64_t, N>& a, const std::array<uint64_t, N>& b)
{
    for (std::size_t i = 0; i < N; ++i)
    {
        a[i] += b[i];
    }
    return a;
}
```
?

Reply:
even better we can define this operator for `ctree<Size, Counters...>::values_t`

Reply:
nevermind, node types are dependent and that interferes with providing `operator+=` for `ctree::values_t`